share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-sles-dpcpp/spack.yaml

```diff
@@ -0,0 +1,64 @@
+spack:
+  include:
+  - $spack/share/spack/gitlab/cloud_pipelines/configs/linux/ci.yaml
+
+  view: false
+
+  concretizer:
+    reuse: false
+    unify: false
+
+  packages:
+    all:
+      require: '%dpcpp'
+      providers:
+        blas: [cray-libsci]
+        lapack: [cray-libsci]
+        mpi: [cray-mpich]
+        tbb: [intel-tbb]
+        scalapack: [netlib-scalapack]
+      target: [zen4]
+      variants: +mpi
+
+    binutils:
+      variants: +ld +gold +headers +libiberty ~nls
+    hdf5:
+      variants: +fortran +hl +shared
+    libunwind:
+      variants: +pic +xz
+    ncurses:
+      require: '@6.3 +termlib'
+    openblas:
+      require: '@0.3.20'
+      variants: threads=openmp
+    xz:
+      variants: +pic
+    elfutils:
+      variants: +bzip2 ~nls +xz
+      require: '%gcc'
+    unzip:
+      require: '%gcc'
+
+  # Only mfem successfully installed in environment equivalent at command line
+  specs:
+  - adios2
+  - amrex
+  - butterflypack
+  - conduit
+  - flux-core
+  - h5bench
+  - hdf5-vol-async
+  - hdf5-vol-cache
+  - hdf5-vol-log
+  - hypre
+  - kokkos
+  - kokkos-kernels
+  - legion
+  - mfem
+  - raja
+  - superlu-dist
+
+  mirrors: { "mirror": "s3://spack-binaries-cray/develop/e4s-cray-sles" }
```

Comment:
Since these don't need to be private any more, you can point to `spack-binaries/develop/{stack_name}`.

Also, the mirror name needs to match the name of the stack, so all of these will need to add their suffix to the mirror name.